pkg/oauthserver/server/csrf/cookie.go

```diff
@@ -3,67 +3,61 @@ package csrf
 import (
 	"net/http"
 
-	"github.com/pborman/uuid"
+	"github.com/openshift/origin/pkg/oauthserver/server/crypto"
 )
 
 type cookieCsrf struct {
-	name     string
-	path     string
-	domain   string
-	secure   bool
-	httponly bool
+	name   string
+	path   string
+	domain string
+	secure bool
 }
 
 // NewCookieCSRF stores random CSRF tokens in a cookie created with the given options.
 // Empty CSRF tokens or tokens that do not match the value of the cookie on the request
 // are rejected.
-func NewCookieCSRF(name, path, domain string, secure, httponly bool) CSRF {
+func NewCookieCSRF(name, path, domain string, secure bool) CSRF {
 	return &cookieCsrf{
-		name:     name,
-		path:     path,
-		domain:   domain,
-		secure:   secure,
-		httponly: httponly,
+		name:   name,
+		path:   path,
+		domain: domain,
+		secure: secure,
 	}
 }
 
 // Generate implements the CSRF interface
-func (c *cookieCsrf) Generate(w http.ResponseWriter, req *http.Request) (string, error) {
+func (c *cookieCsrf) Generate(w http.ResponseWriter, req *http.Request) string {
+	// reuse the session cookie if we already have one
+	// this makes us more tolerant of multiple clicks against the login page
 	cookie, err := req.Cookie(c.name)
 	if err == nil && len(cookie.Value) > 0 {
-		return cookie.Value, nil
+		return cookie.Value
 	}
 
+	// do not set Expires or MaxAge to make this a session cookie
 	cookie = &http.Cookie{
 		Name:     c.name,
-		Value:    uuid.NewUUID().String(),
+		Value:    crypto.Random256BitsString(),
```

Comment:
just wondering if this change should be part of a refactor rather than its own thing

Reply:
If the UUID bits had be a real security issue I would have made a separate PR with the one line change to the correct UUID function.